ibm/data_source_ibm_is_security_group_rule.go

```diff
@@ -0,0 +1,304 @@
+						},
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier for this security group.",
+						},
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The user-defined name for this security group. Names must be unique within the VPC the security group resides in.",
+						},
+					},
+				},
+			},
+			"code": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Optional:    true,
+				Description: "The ICMP traffic code to allow.",
+			},
+			"type": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Optional:    true,
+				Description: "The ICMP traffic type to allow.",
+			},
+			"port_max": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Optional:    true,
+				Description: "The inclusive upper bound of TCP/UDP port range.",
+			},
+			"port_min": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Optional:    true,
+				Description: "The inclusive lower bound of TCP/UDP port range.",
+			},
+		},
+	}
+}
+
+func dataSourceIBMIsSecurityGroupRuleRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcClient, err := meta.(ClientSession).VpcV1API()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getSecurityGroupRuleOptions := &vpcv1.GetSecurityGroupRuleOptions{}
+
+	getSecurityGroupRuleOptions.SetSecurityGroupID(d.Get("security_group_id").(string))
+	getSecurityGroupRuleOptions.SetID(d.Get("id").(string))
+
+	securityGroupRuleIntf, response, err := vpcClient.GetSecurityGroupRuleWithContext(context, getSecurityGroupRuleOptions)
+	if err != nil {
+		log.Printf("[DEBUG] GetSecurityGroupRuleWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("GetSecurityGroupRuleWithContext failed %s\n%s", err, response))
+	}
+
+	switch reflect.TypeOf(securityGroupRuleIntf).String() {
+	case "*vpcv1.SecurityGroupRuleSecurityGroupRuleProtocolAll":
+		{
+			securityGroupRule := securityGroupRuleIntf.(*vpcv1.SecurityGroupRuleSecurityGroupRuleProtocolAll)
+
+			d.SetId(*securityGroupRule.ID)
+			if err = d.Set("direction", securityGroupRule.Direction); err != nil {
+				return diag.FromErr(fmt.Errorf("Error setting direction: %s", err))
+			}
+			if err = d.Set("href", securityGroupRule.Href); err != nil {
+				return diag.FromErr(fmt.Errorf("Error setting href: %s", err))
+			}
+			if err = d.Set("ip_version", securityGroupRule.IPVersion); err != nil {
+				return diag.FromErr(fmt.Errorf("Error setting ip_version: %s", err))
+			}
+			if err = d.Set("protocol", securityGroupRule.Protocol); err != nil {
+				return diag.FromErr(fmt.Errorf("Error setting protocol: %s", err))
+			}
+			if securityGroupRule.Remote != nil {
+				v, err := dataSourceSecurityGroupRuleFlattenRemote(securityGroupRule.Remote)
```

Comment:
rename 'v' to 'securityGroupRuleRemote' or 'remote'

Reply:
done